### ISSUE TITLE
[Bug]: Record of Deleted Mod persists preventing reinstallation

### ISSUE THREAD

**Issue by Rinual** [Kizari/Flagrum#167]
### Operating System

Windows 10 64-Bit

### Game Version

Latest Steam Release

### Version

1.5.14

### Description of the bug

- After manually merging mods, keeping two separate windows of Flagrum open to keep records of source and target mod, I would delete the source mod. 
- Mods that have been deleted, reinstalling the mod will not pop-up the window saying an existing mod is already installed, however flagrum would crash, with the given key being already present. 
- Flagrum will even re-create the .fproj folder, however even with this recreated .fproj folder, or by manually deleting the folder, flagrum will still crash with the same error from an existing key

### Steps to reproduce the bug

1) I'm not up for testing to see how replicable this is, as its likely to just mess up flagrums more, but I assume this happened from having two versions of flagrum up at the same time while deleting a mod. 

### Crash logs

[2024-10-10_12-30-47-4811_crash_1959007222.txt](https://github.com/user-attachments/files/17331492/2024-10-10_12-30-47-4811_crash_1959007222.txt)

### Additional context

I assume the GUID must be getting properly errased, which is why Flagrum isn't detecting the conflict, but 
An item with the same key has already been added. Key: abbc531c-4e40-45c9-b7f7-9cefa720107a
Implies there is a 2nd unique identifier that doesn't have an exception being thrown that needs a way of getting either manually cleaned or automatically cleaned

### Code of Conduct

- [X] I confirm that I have ensured this bug hasn't already been reported in another issue on this repository.
- [X] I confirm that this bug report is pertaining to Flagrum, not an individual mod, the game itself, or an issue with my system or files.
- [X] I confirm that I am using a legitimate copy of the game.
- [X] I confirm that I installed Flagrum with a clean installation of the game, and have not used any tools other than Flagrum to mod my game files.
- [X] If my issue relates to the game itself crashing or exhibiting unexpected behaviour, I confirm that I have tried running a clean unmodded installation of the game to see if this resolves my issue before filing a bug report.

**Comment by Kizari** [comment]
It's hard to say what went wrong here since Flagrum really wasn't designed with multiple instances in mind. I'm assuming the app state became out of sync between the two due to changing in one but not the other (such as deleting the mod), but then the incorrect app state was saved due to an action taken in that instance.

Under the relevant profile's folder, there's a file called `mod_state.zstd` which holds state information pertaining to your mods. This information in this file I believe just holds data for each mod such about whether it is currently enabled, and whether it is currently pinned. As such, if you just disable all mods, you should just be able to close Flagrum, delete this file, relaunch Flagrum, then repin the mods that you wanted pinned. **Copy it somewhere else first in case I've forgotten something and it makes the situation worse.**

**Comment by Rinual** [comment]
Yep, that does fix that issue. Good enough of a solution for me, though if anyone is repeating this, be aware it will reset all your pinning of mods. But that shouldn't be a big deal to anyone besides me, but this likely wouldnt happen to anyone besides me either. 
Thanks for the new info, it's exactly what i was hoping for.

**Comment by Rinual** [comment]
Just going to close this, as this is a perfect solution.
